src/core/core.js

```diff
@@ -370,6 +377,13 @@ export default class Core {
       })
       .then(response => SearchDataTransformer.transformUniversal(response, urls, this._fieldFormatters))
       .then(data => {
+        const previousQueryId = this.storage.get(StorageKeys.QUERY_ID);
```

Comment:
can we consolidate this logic somewhere? maybe a shared function at the bottom of the file? (I have no clue where else this would live)